repo.go

```diff
@@ -309,19 +309,33 @@ func (r *Repo) GenKey(role string) ([]string, error) {
 }
 
 func (r *Repo) GenKeyWithExpires(keyRole string, expires time.Time) (keyids []string, err error) {
-	// Not compatible with delegated targets roles, since delegated targets keys
-	// are associated with a delegation (edge), not a role (node).
+	return r.GenKeyWithSchemeAndExpires(keyRole, expires, data.KeySchemeEd25519)
+}
 
-	signer, err := keys.GenerateEd25519Key()
+func (r *Repo) GenKeyWithSchemeAndExpires(role string, expires time.Time, keyScheme data.KeyScheme) ([]string, error) {
```

Comment:
I'm sensing combinatorial explosion...what if I want a custom scheme but the default expiration time?

What do Go developers do in this case? (TBH I would believe you if you told me "write 2^(# options) functions 😛 )

Reply:
I think you could just use ` data.DefaultExpires(role)`? We may want to expose a `data.DefaultScheme` but I don't think it's necessary given the function sigs.

Reply:
Yeah, I'm wondering if maybe something like this (from Rob Pike) would be appropriate: https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html

Or maybe an options struct (number 2 [here](https://golang.cafe/blog/golang-functional-options-pattern.html)

(Totally fine to punt on this for now)